src/skibidi_orm/migration_engine/db_inspectors/db_inspector.py

```diff
@@ -0,0 +1,127 @@
+from typing import Any, Literal, cast
+import sqlite3
+from skibidi_orm.migration_engine.adapters.sqlite3_adapter import SQLite3Adapter
+from skibidi_orm.migration_engine.db_config.sqlite3_config import SQLite3Config
+from skibidi_orm.migration_engine.db_inspectors.base_inspector import BaseDbInspector
+
+
+type SQLite3PragmaTableInfo = list[
+    tuple[int, str, str, Literal[0, 1], Any, Literal[0, 1]]
+]
+
+type SQLite3PragmaForeignKeyListEntry = tuple[
+    int,
+    int,
+    str,
+    str,
+    str,
+    Literal["NO ACTION", "RESTRICT", "SET NULL", "SET DEFAULT", "CASCADE"],
+    Literal["NO ACTION", "RESTRICT", "SET NULL", "SET DEFAULT", "CASCADE"],
+    str,
+]
+
+type SQLite3PragmaForeignKeyList = list[SQLite3PragmaForeignKeyListEntry]
+
+type SQLite3PragmaForeignKeyListWithTableName = tuple[
+    SQLite3PragmaForeignKeyList,
+    str,
+]
+
+
+class SqliteInspector(BaseDbInspector):
+    """
+    Used to get data from live SQLite3 database.
+    Should only be instantiated when SQLite3 is choosen as the database.
+    """
+
+    def __init__(self) -> None:
+        self.config = SQLite3Config.instance()
+
+    def get_tables(
+        self,
+    ) -> list[SQLite3Adapter.Table]:
+        tables: list[SQLite3Adapter.Table] = []
+        tables_names = self.get_tables_names()
+        for table_name in tables_names:
+            table_columns = self.get_table_columns(table_name)
+            tables.append(SQLite3Adapter.Table(name=table_name, columns=table_columns))
+
+        return tables
+
+    def get_tables_names(self) -> list[str]:
+        tables = self._sqlite_execute(
+            "SELECT name FROM sqlite_master WHERE type='table';"
+        )
+        return [table[0] for table in tables]
+
+    def get_relations(self) -> list[SQLite3Adapter.Relation]:
+        tables_names = self.get_tables_names()
+        foreign_keys: list[SQLite3PragmaForeignKeyListWithTableName] = list(
+            filter(
+                lambda x: x[0],
+                [
+                    (self._sqlite_execute(f"PRAGMA foreign_key_list({table})"), table)
+                    for table in tables_names
+                ],
+            )
+        )
+
+        relations: list[SQLite3Adapter.Relation] = []
+        for (
+            relation_list,
+            origin_table_name,
+        ) in foreign_keys:
+            for (
+                _,
+                _,
+                referenced_table,
```

Comment:
imo we should write some wrapper for it instead of ignoring 5/8 returned values, maybe some dataclass that returns a list of those 3 through some method? maybe just ignore the rest during the init? aint no way we're gonna need some of those values like `hard_heap_limit`